aten/src/ATen/native/sparse/SparseBinaryOpIntersectionCommon.h

```diff
@@ -360,12 +360,12 @@ void _sparse_binary_op_intersection_kernel_impl(
       KernelLauncher::launch(iter,
           // NOTE: capture by value required by CUDA
           [=] FUNCAPI (index_t nnz_idx) -> index_t {
-          // Compute hash value
-          const auto* RESTRICT ptr_indices_dim = ptr_indices + nnz_idx * indices_nnz_stride;
           int64_t hash = 0;
           if (hash_ptr) {
             hash = hash_ptr[nnz_idx];
-          } else {
+          } else if (sparse_dim) {
+            // Compute hash value
+            const auto* RESTRICT ptr_indices_dim = ptr_indices + nnz_idx * indices_nnz_stride;
```

Comment:
nit: Hmm, could you please add a comment about why this condition? Otherwise it is not very clear...
Also, what happens when `sparse_dim == 0`?

Reply:
Nay, `sparse_dim` can be zero for empty tensors, when the entire loop below is a no-op but it triggers the UBSan error, as adding non-zero offset to `ptr_indices`, which are null in that case is bogus.

Will add the comment to PR description

Reply:
@malfet, are you sure? The constructor errors when `sparse_dim < 1` and the following:
```python
>>> x = torch.rand(0)
>>> x
tensor([])
>>> xs = x.to_sparse()
>>> xs
tensor(indices=tensor([], size=(1, 0)),
       values=tensor([], size=(0,)),
       size=(0,), nnz=0, layout=torch.sparse_coo)
>>> xs.sparse_dim()
1
```
If I am missing something, I would love to see a counter-example :)

Reply:
Ah, never mind, got it here:
```python
>>> indices = torch.rand(0, 0).to(torch.long)
>>> values = torch.rand(0)
>>> torch.sparse_coo_tensor(indices, values, ())
tensor(indices=tensor([], size=(0, 0)),
       values=tensor([], size=(0,)),
       size=(), nnz=0, layout=torch.sparse_coo)
>>> torch.sparse_coo_tensor(indices, values, ()).sparse_dim()
0
```
Sorry for the noise :)